File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        p = ""
        for j in range(len(strs[0])):
            c = strs[0][j]
            for st in strs:
                if len(st) <= j or st[j] != c:
                    return p
            p += c
        return p

File: 0014-longest-common-prefix/0014-longest-common-prefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
    p = ""
    for i in 0..strs[0].length() - 1
        c = strs[0][i]
        strs.each do |st|
            if st.length() <= i or st[i] != c
                return p
            end
        end
        p += c
    end
    return p
end